sdk/dotnet/Pulumi/Deployment/IMonitor.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+using Pulumirpc;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// Abstraction for hooks to the engine and monitor.
+    /// </summary>
+    internal interface IMonitor
```

Comment:
See above, i would call this somethin glike `IRpc` or `IRpcDispatcher`  Then you can have your `MockRpcDispatcher` and `GrpcDispatcher`. 

From a regular design perspective i somewhat like that as it makes it very clear in our system at which point we're calling out through rpc to some other component.